SConstruct

```diff
@@ -107,22 +107,10 @@ for x in sorted(glob.glob("platform/*")):
     sys.path.remove(tmppath)
     sys.modules.pop("detect")
 
-custom_tools = ["default"]
-
-platform_arg = ARGUMENTS.get("platform", ARGUMENTS.get("p", False))
-
-if platform_arg == "android":
-    custom_tools = ["clang", "clang++", "as", "ar", "link"]
-elif platform_arg == "web":
-    # Use generic POSIX build toolchain for Emscripten.
-    custom_tools = ["cc", "c++", "ar", "link", "textfile", "zip"]
-elif os.name == "nt" and methods.get_cmdline_bool("use_mingw", False):
-    custom_tools = ["mingw"]
```

Comment:
Change the order of this stuff is pretty risky and regression prone. We've been bitten by this multiple times, we'd need to be extremely cautious here. That's sadly too risky for a bugfix PR IMO, we should go back to the previous working stage if that's the only option.

Reply:
Could you investigate instead whether https://github.com/godotengine/godot/pull/99762 can be used as a base to solve the issue in a better way? I'm more confident in that change, it's just a draft because I wanted to do more testing.

Reply:
I suppose I could turn this on its head by making the platform *evaluation* happen earlier instead of the tool assignment. That's certainly more isolated, so it wouldn't have that same risk of regression. Gonna give that a shot

That PR *could* be a base, but at the same time I think a lot of the platform/module specific setup should be reimplemented as SCons-native tools. Gonna have to whip up a proposal at some point that consolidates a lot of these ideas for further discussion

Reply:
In second thought I think there's merit in doing this change indeed, the current logic that relies on querying `ARGUMENTS` to know the selected platform before having `env["platform"]` is very brittle.

I'd like to combine this PR and #99762 somehow and see if we can make it safe.

Reply:
I agree there's merit to it, but now I'm not sold on it being worth digging into in isolation. There's *plenty* of areas here that use `ARGUMENTS` incorrectly, and trying to address all of them would go out of scope